### ISSUE TITLE
Show snapshot diffs in files list

### ISSUE THREAD

**Issue by emuell** [emuell/restic-browser#9]
There could be a new "diff" mode, where two snapshots can be selected and the file list then shows the changes files with removed, added, changed tags.

Ideally the diff file list also should be a browsable tree and not a flat list like output of "restic diff". Unfortunately the restic diff command does not support path filtering, so the entire change list first must be queried before the changes can be shown.

**Comment by damoclark** [comment]
Hi @emuell 

You could add to this functionality, a diff with what is present on the current filesystem location also.  

So ability to compare what's in a specific snapshot against what is currently stored on the filesystem.  For a restoration focused tool, I think this might be more useful when looking to find the appropriate snapshot to restore from, than simply comparing backup snapshots with one another.

Just a thought.

Damo.

**Comment by emuell** [comment]
Basically what https://github.com/bit-team/backintime does with its "now" location?

That's definitely a good idea, but it requires an additional setup. I personally use restic for my local machine backups, but also for a lot of other remote machines. So in order to browse the actual state of those remote machines, you'd first need to tell restic-browser first how to access that via for example rclone or ssh - additionally to the restic repository location.

Ideally this then also should be saved in some form of a *preset*, so you don't have to specify this again and again every time you open restic browser.

**Comment by damoclark** [comment]
> Basically what https://github.com/bit-team/backintime does with its "now" location?

Yes.  They also have a feature where you can perform diffs between files in snapshots as well as 'now' on the filesystem, but that's something different to this issue.

> I personally use restic for my local machine backups, but also for a lot of other remote machines. So in order to browse the actual state of those remote machines, you'd first need to tell restic-browser how to access that location via for example rclone or ssh - additionally to the restic repository location.

I hadn't considered the use-case that you describe.  To simplify things, at least initially, the feature I described might be one relevant only when you are looking to restore files, and running Restic-browser on the host you wish to restore to.  

Attempting to make it work from a host that doesn't contain the actual files to compare with might not be worth the complexity.

Alternately, you may attempt to remotely mount the destination for the restore on the host running Restic-browser, where you might need to specify the path location, relative to the root path of the backup.
